From 8e0a3e569e26a5c5812d72d026b88f3347b519ae Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Thu, 1 Nov 2018 16:56:49 +0800
Subject: [PATCH 5072/5242] MLK-20302-2 ARM64: dts: imx8qm: add rfkill for
 domu car

commit  486074baf25e03872b879f7992def8eac6657436 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add rfkill node for domu car for bluetooth usage.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Reviewed-by: Flynn xu <flynn.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-mek-dom0.dts     |    1 +
 .../boot/dts/freescale/fsl-imx8qm-mek-domu-car.dts |    6 ++++++
 2 files changed, 7 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
index af61cbe..9822e0d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
@@ -234,6 +234,7 @@
 				<&gpio4 9 GPIO_ACTIVE_LOW>,
 				<&gpio4 11 GPIO_ACTIVE_HIGH>,
 				<&gpio4 19 GPIO_ACTIVE_HIGH>,
+				<&gpio4 22 GPIO_ACTIVE_LOW>,
 				<&gpio4 26 GPIO_ACTIVE_HIGH>,
 				<&gpio4 25 GPIO_ACTIVE_HIGH>,
 				<&gpio4 27 GPIO_ACTIVE_LOW>,
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu-car.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu-car.dts
index 9783947..8021349 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu-car.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu-car.dts
@@ -34,6 +34,12 @@
 		#interrupt-cells = <1>;
 	};
 	*/
+
+	bt_rfkill {
+		compatible = "fsl,mxc_bt_rfkill";
+		bt-power-gpios = <&gpio4 22 GPIO_ACTIVE_LOW>;
+		status ="okay";
+	};
 };
 
 &iomuxc {
-- 
1.7.9.5

